### ISSUE TITLE
JDBCSessionDataStore doesn't work with root context on Oracle DB

### ISSUE THREAD

**Issue by jamesagnew** [eclipse/jetty.project#1591]
JDBCSessionDataStore creates a NOT NULL column called `CONTEXT_PATH` in the session table.

When persisting a session against the root context, the server inserts a row with an empty string in this column. This works on Postgres and MySQL, but Oracle treats the empty string as a null and refuses to persist it:

```
java.sql.SQLIntegrityConstraintViolationException: ORA-01400: cannot insert NULL into ("TESTDB"."JETTY_SESSION"."CONTEXT_PATH")
```

**Comment by janbartel** [comment]
Which version of jetty is this? I'm fairly sure this has been tested with newer versions of jetty, but may have been a problem in an older version, or perhaps it is a regression.

**Comment by jamesagnew** [comment]
Hi @janbartel, sorry, I should have included that detail.

This is happening for me on `9.4.3.v20170317`, and I've also reproduced it on `9.4.5.v20170502`. I did not have this issue on the 9.2.x series. 

I suspect this is a regression in 9.4- there is definitely special case code for Oracle's weird empty string handling (e.g. [here](https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-server/src/main/java/org/eclipse/jetty/server/session/JDBCSessionDataStore.java#L441) but since 9.4 introduced code to make the PK of the table be a combination of the ID, Context and VHost it's no longer possible for that column to be 'null' in Oracle so this special case could never be hit.

**Comment by janbartel** [comment]
@jamesagnew I pushed a change, any chance you could check out jetty-9.4.x branch, build and test against it?

**Comment by jamesagnew** [comment]
Fantastic, sure thing I'll give it a shot later today! Thanks!

**Comment by jamesagnew** [comment]
This seems like it does the trick. Thanks for the quick response!

**Comment by janbartel** [comment]
Thanks @jamesagnew. Closing as fixed for jetty-9.4.x.
